js/ui/appIconBar.js

```diff
@@ -1004,6 +1036,23 @@ const AppIconBar = new Lang.Class({
         Main.overview.connect('showing', Lang.bind(this, this._updateActiveApp));
         Main.overview.connect('hidden', Lang.bind(this, this._updateActiveApp));
 
+
+        for (let index = 0; index < 8; index++) {
+            let fullName = 'activate-icon-' + (index + 1);
+            Main.wm.addKeybinding(fullName,
+                                  new Gio.Settings({ schema: SHELL_KEYBINDINGS_SCHEMA }),
+                                  Meta.KeyBindingFlags.NONE,
+                                  Shell.KeyBindingMode.NORMAL |
+                                  Shell.KeyBindingMode.OVERVIEW,
+                                  this._activateNthApp.bind(this, index));
```

Comment:
You could pass the index to `Lang.bind()`, but you would have to change the prototype of it to include the arguments that the keybinding handler function includes...
Another way to do so is to declare a function inline here that calls into `this._activateNthApp()`.